bee-message/src/payload/transaction/essence/regular.rs

```diff
@@ -26,10 +26,13 @@ pub struct RegularTransactionEssenceBuilder {
 
 impl RegularTransactionEssenceBuilder {
     /// Creates a new [`RegularTransactionEssenceBuilder`].
-    pub fn new(network_id: u64) -> Self {
+    pub fn new(network_id: u64, inputs_commitment: [u8; 32]) -> Self {
         Self {
             network_id,
-            ..Default::default()
+            inputs: Vec::new(),
+            inputs_commitment,
```

Comment:
Why not just add `inputs_commitment` and keep the rest `..Default::default()`?

Reply:
No real reason, personal preference 😅